index.js

```diff
@@ -43,7 +23,13 @@ module.exports = function(options){
 
     if(file.isBuffer()){
       try {
-        file.contents = new Buffer(handleCompile(String(file.contents), opts));
+        var compiled;
+        if (opts.client) {
+          compiled = jade.compileClient(String(file.contents), opts);
```

Comment:
do the `String(file.contents)` before the `if` and assign to variable, keep it dry.